### DISCUSSION TITLE
How to use existing openjdk-17 in Windows?

### DISCUSSION THREAD

**Discussion by skaveesh** [nvim-java/nvim-java#206] [Q&A] [answered]
My `init.lua` file look like this. And this does not work :
<sub>(I'm still trying to figure out nvim configs. This is my day-2)</sub>
```lua
-- removed unrelated config
require('lazy').setup({
-- removed unrelated config
{
    'nvim-java/nvim-java',
    dependencies = {
      'nvim-java/lua-async-await',
      'nvim-java/nvim-java-refactor',
      'nvim-java/nvim-java-core',
      'nvim-java/nvim-java-test',
      'nvim-java/nvim-java-dap',
      'MunifTanjim/nui.nvim',
      'neovim/nvim-lspconfig',
      'mfussenegger/nvim-dap',
      {
        'williamboman/mason.nvim',
        opts = {
          registries = {
            'github:nvim-java/mason-registry',
            'github:mason-org/mason-registry',
          },
        },
      }
    },
  },
-- removed unrelated config
})

require('java').setup({
  java_home = "C:\\Program Files\\Java\\jdk-17"
})

require('lspconfig').jdtls.setup({
  settings = {
    java = {
      configuration = {
        runtimes = {
          {
            name = "JavaJDK-17",
            path = "C:\\Program Files\\Java\\jdk-17",
            default = true,
          }
        }
      }
    }
  }
})
```

With this configuration if I start `nvim` it automatically start downloading `openjdk-17`. I don't want another copy of JDK-17 downloading to my PC. How do I use the existing jdk-17 (Windows 10, I'm not using WSL for nvim)? 

I even tried without whitespace in the path variable. But no luck:
```
path = "C:\\PROGRA~1\\Java\\jdk-17",
```

As below, it starts pulling a new copy of openjdk-17:
![WindowsTerminal_ODm8wzK13u](https://github.com/nvim-java/nvim-java/assets/14146760/f2676e9e-981d-431f-b97b-b867aa79bd8f)

**Comment by s1n7ax** [comment, marked as answer]
@skaveesh Set jdk `auto_install` to `false` in config. You cannot pass arbitrary properties like `java_home` to plugin setup and expect it to work. Since you are new, Almost 90% of lua neovim plugins has configuration information in the README. Here is this plugin's configuration. All the properties I have defined are the default meaning If you are fine with the default, just call setup without any parameters. `require('java').setup()`
 
https://github.com/nvim-java/nvim-java?tab=readme-ov-file#wrench-configuration

However, in your case, you want to disable openjdk installation. So, calling setup like below will override `auto_install` but use rest of the default parameters as it is.

```
require('java').setup({
 jdk = {
  auto_install = false,
 },
})
```

Feel free to reopen if you have any issues.

**Reply by skaveesh** [reply]
Thanks for the help. I was following the [🗜️ How to Use JDK X.X Version?](https://github.com/nvim-java/nvim-java?tab=readme-ov-file#clamp-how-to-use-jdk-xx-version) but did not understand the configurations under 🔧 Configuration or where to put them. Then was coping bunch of configs on the internet cluelessly. Thanks again
